### PR TITLE
UI - Fixes modal width by device screen

### PR DESCRIPTION
**Pull request by utchoang**
### Description

Fixes https://github.com/apache/cloudstack/issues/5524

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### Screenshots (if appropriate):
After fixes:
image

### CODE REVIEW

**Comment by utchoang** [comment]
@blueorangutan ui

File: ui/src/views/AutogenView.vue [hunk 1]
```diff
@@ -1524,13 +1527,18 @@ export default {
       } else {
         callback()
       }
+    },
+    setModalWidthByScreen () {
+      const screenWidth = window.innerWidth
+      if (screenWidth <= 768) {
+        this.modalWidth = '450px'
```
Comment (rhtyd) [inline ui/src/views/AutogenView.vue:1534]: Should this be like a percentage of total width?

**Comment by rhtyd** [comment]
@utchoang lgtm on my phone, Ill

**Comment by utchoang** [comment]
> @utchoang lgtm on my phone, I'll need to test a few more screen sizes. The reboot VM button on mobile for me doesn't work, can you check that?

As far as I found this error comes from the Api `rebootVirtualMachine` no longer having the `force` parameter. Is it ok if I remove it from the reboot form?
image

File: ui/src/views/AutogenView.vue (hunk 1 above)
Comment (utchoang) [reply ui/src/views/AutogenView.vue:1534]: No, currently I only hard fix the width for ipad, tablet screens and below. Do you feel it is ok or do you need to follow the percentage of the screen?

**Comment by rhtyd** [comment]
Ah okay, no keep it @utchoang I found the forced parameter was added in 4.16  but qa.cloudstack.org server/backend is 4.15.2.0. I'll upgrade its backend to 4.16-snapshot shortly.

**Review by rhtyd — APPROVED** [review]

**Comment by rhtyd** [comment]
@utchoang can you check and update/fix the UI unit tests. This is as seen from Travis's job 1:
```
 cloudstack-ui@1.0.0 test:unit /home/travis/build/apache/cloudstack/ui
> vue-cli-service test:unit
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
[vue-jest]: Less are not currently compiled by vue-jest
 FAIL  tests/unit/views/AutogenView.spec.js (33.139s)
....
```

**Comment by utchoang** [comment]
@blueorangutan ui

**Review by borisstoyanov — CHANGES_REQUESTED** [review]
Screenshot 2021-09-29 at 13 55 57
hi @utchoang thanks for this fix, currently the action list overlaps the dialogue window.

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by rhtyd** [comment]
@blueorangutan ui

**Review by nvazquez — APPROVED** [review]

**Comment by nvazquez** [comment]
@rhtyd @borisstoyanov please confirm if your comments have been addressed and if its ready to merge, thanks

File: ui/src/views/AutogenView.vue (hunk 1 above)
Comment (davidjumani) [reply ui/src/views/AutogenView.vue:1533]: I too think percentage will be better. Perhaps 90% ?

**Comment by rhtyd** [comment]
Tested lgtm

**Comment by borisstoyanov** [comment]
overlap
Thanks @utchoang, but now the tooltip stays on top of the window after you press the button

**Comment by utchoang** [comment]
@borisroman In my opinion, it is ant design behavior so blocking it will cause conflict with the tooltip in the web interface. Refer: https://www.antdv.com/components/tooltip/#Tooltip

**Pull request merged**